src/relayFeeCalculator/chain-queries/baseQuery.ts

```diff
@@ -93,7 +94,12 @@ export class QueryBase implements QueryInterface {
       transport,
     } = options;
 
-    const tx = await this.getUnsignedTxFromDeposit(deposit, relayer);
+    const { recipient, outputToken, exclusiveRelayer } = deposit;
+    assert(recipient.isEVM());
+    assert(outputToken.isEVM());
+    assert(exclusiveRelayer.isEVM());
+
+    const tx = await this.getUnsignedTxFromDeposit({ ...deposit, recipient, outputToken, exclusiveRelayer }, relayer);
```

Comment:
Ideally this wouldn't be needed, but I think we can live with it for now.

A better way would be to update the `RelayFeeCalculator` interface to permit a union of types for `getGasCosts()` - i.e. such that implementations can choose to implement _either_ the EVM variant or SVM variant of the fill. There's precedent for doing that in the `GasPriceOracle` RE EVM/SVM gas price estimates.

Reply:
I think this is fine if we want to get this code running and be backwards compatible and then fast follow by removing that seems like a solid plan